seed/ts-sdk/endpoint-security-auth/src/auth/RoutingAuthProvider.ts

```diff
@@ -0,0 +1,80 @@
+// This file was auto-generated by Fern from our API Definition.
+
+import type * as core from "../core/index.js";
+import type { BearerAuthProvider } from "./BearerAuthProvider.js";
+import type { HeaderAuthProvider } from "./HeaderAuthProvider.js";
+import type { OAuthAuthProvider } from "./OAuthAuthProvider.js";
+
+export namespace RoutingAuthProvider {
+    type UnionToIntersection<U> = (U extends any ? (x: U) => void : never) extends (x: infer I) => void ? I : never;
+
+    type AtLeastOneOf<T extends any[]> = {
+        [K in keyof T]: T[K] & Partial<UnionToIntersection<Exclude<T[number], T[K]>>>;
+    }[number];
+
+    export type AuthOptions = AtLeastOneOf<
```

Comment:
Some endpoints may be anonymous, so we should not enforce `AtLeastOneOf` in this case.